### PR TITLE
Clean up javadoc for aerogear-sync-server

### PR DESCRIPTION
**Pull request by danbev**
To test:
1) Run the following command from the root directory:
```shell
mvn clean install javadoc:aggregate
```
2) Open ```target/site/apidocs/index.html``` in a web browser and inspect the content.

JIRA: [AGSYNC-32](https://issues.jboss.org/browse/AGSYNC-32)

### CODE REVIEW

File: api/src/main/java/org/jboss/aerogear/sync/Diff.java [hunk 1]
```diff
@@ -16,5 +16,14 @@
  */
 package org.jboss.aerogear.sync;
 
+import org.jboss.aerogear.sync.client.ClientSynchronizer;
+import org.jboss.aerogear.sync.server.ServerSynchronizer;
+
+/**
+ * A marker interface that represents a diff or two versions of a document/object.
+ * <p>
+ * The actual implementation of a diff will vary depending on the type of content the the
```
Comment (cvasilak) [inline api/src/main/java/org/jboss/aerogear/sync/Diff.java]: s/ the the/ the

File: api/src/main/java/org/jboss/aerogear/sync/DataStore.java [hunk 2]
```diff
@@ -22,7 +22,8 @@
  * A DataStore implementation is responible for storing and serving data for a
```
Comment (cvasilak) [inline api/src/main/java/org/jboss/aerogear/sync/DataStore.java]: s/responible/responsible

**Comment by cvasilak** [comment]
on interface ```ShadowDocument``` maybe we rephrase:
s/A shadow document is update after a successful patch/merge as been performed
s/A shadow document is updated after a successful patch/merge is executed.

or something like that..

**Comment by cvasilak** [comment]
apart from this small typos looks good :+1:

**Comment by danbev** [comment]
@cvasilak Thanks, I've update the PR. Let me know if this reads better now.

**Comment by cvasilak** [comment]
:+1: looks good

**Comment by matzew** [comment]
:+1:

**Pull request merged**